pdns/pdns_recursor.cc

```diff
@@ -1172,9 +1179,24 @@ static void startDoResolve(void *p)
             // Option Code and Option Length are both 2
             maxanswersize -= 2 + 2 + mode_server_id.size();
           }
+        } else if (paddingAllowed && padResponse == false && g_paddingMode == PaddingMode::PaddedQueries && o.first == EDNSOptionCode::PADDING) {
+          /* we should only pad on 'secure' cases to limit amplification:
+             - over TCP ;
+             - from 'known-good' sources (dnsdist for example).
+             We should also support a strict mode (only pad if the query has a padding option)
```

Comment:
```suggestion
             TODO: We should also support a strict mode (only pad if the query has a padding option)
```

Reply:
It's actually already implemented, I'll update the comment :)